api/flags/config.go

```diff
@@ -0,0 +1,45 @@
+package flags
+
+import (
+	"fmt"
+
+	common "github.com/base-org/blob-archiver/common/flags"
+	oplog "github.com/ethereum-optimism/optimism/op-service/log"
+	opmetrics "github.com/ethereum-optimism/optimism/op-service/metrics"
+	"github.com/urfave/cli/v2"
+)
+
+type APIConfig struct {
+	LogConfig     oplog.CLIConfig
+	MetricsConfig opmetrics.CLIConfig
+	BeaconConfig  common.BeaconConfig
+	StorageConfig common.StorageConfig
+
+	ListenAddr string
+}
+
+func (c APIConfig) Check() error {
+	if err := c.StorageConfig.Check(); err != nil {
+		return err
```

Comment:
Probably worth wrapping the error here & on L27 so it's easy to distinguish these two cases should either arise.